lib/active_admin/middleware/error_handler.rb

```diff
@@ -0,0 +1,54 @@
+module ActiveAdmin
+  class ErrorHandler
+
+    def initialize(app)
+      @app = app
+    end
+
+    def call(env)
+      @env = env
+      @app.call(env)
+    rescue Exception => exception
+      if is_active_admin_error? && use_active_admin_error_page?
+        render_exception(exception)
+      else
+        raise exception
+      end
+    end
+
+    private
+
+    def is_active_admin_error?
+      stringified_namespaces.include?(current_namespace)
+    end
+
+    def use_active_admin_error_page?
+      if ActiveAdmin.application.use_active_admin_error_page.is_a?(Proc)
+        ActiveAdmin.application.use_active_admin_error_page.call
+      else
+        ActiveAdmin.application.use_active_admin_error_page
+      end
+    end
+
+    def render_exception(exception)
+      @env["active_admin.original_error"] = exception
+
+      @env["action_dispatch.request.parameters"] =
+          {controller: "#{current_namespace}/error", action: "index"}
```

Comment:
If I don't put in line 36, the `AbstractController::ActionNotFound` exception will be raised by the base controller, for every errored action that is not index to its corresponding controller.


Here I'm modifying the `params` hash to make the `BaseController`'s  `before_filter :only_render_implemented_actions` understand that the request is coming from the `index` action of `Admin::ErrorController`

The cleaner way would have been to create a new `Rack::Request` object and do a `update_param`. But this was not working. So, I did it this way